cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -150,6 +152,43 @@ public MasterServiceMain() {
     this.serviceStore = injector.getInstance(ServiceStore.class);
     this.secureStoreUpdater = baseInjector.getInstance(TokenSecureStoreUpdater.class);
     this.leaderElection = createLeaderElection();
+
+    this.masterServices = new RetryOnStartFailureService(new Supplier<Service>() {
+      @Override
+      public Service get() {
+        return new AbstractIdleService() {
+          @Override
+          protected void startUp() throws Exception {
```

Comment:
So all of the operations in `startUp` can be executed multiple times without a corresponding call to `shutDown`?